ui/index.html

```diff
@@ -1401,34 +1380,58 @@ <h5 class="modal-title">Connect wallet</h5>
                     // approve disabled, cross tokens enabled
                     disableApproveCross({
                         approvalDisable: true,
-                        doNotAskDisabled: true,
                         crossDisabled: true
                     });
 
-
-                    return new Promise((resolve, reject) => {
-
-                        tokenContract.methods.approve(bridgeContract.options.address, amountBN.mul(new BN(101)).div(new BN(100)).toString())
-                            .send({from: address, gasPrice:gasPrice, gas:70_000}, async (err, txHash) => {
-                            if (err) return reject(err);
-                                try {
-                                let receipt = await waitForReceipt(txHash);
-                                if(receipt.status) {
-                                    resolve(receipt);
+                    return Promise.resolve().then(() => {
+                        return new Promise(async (resolve, reject) => {
+                            if (config.networkId == 31 && token.token === 'USDT') {
+                                const allowance = await retry3Times(tokenContract.methods.allowance(address, bridgeContract.options.address).call);
+                                if (allowance.toString() != '0') {
+                                    // To change the approve amount you first have to reduce the addresses
+                                    // allowance to zero by calling `approve(_spender, 0)` for USDT
+                                    // https://etherscan.io/address/0xdac17f958d2ee523a2206206994597c13d831ec7#code
+                                    tokenContract.methods.approve(bridgeContract.options.address, '0')
+                                        .send({from: address, gasPrice:gasPrice, gas:70_000}, async (err, txHash) => {
+                                        if (err) return reject(err);
+                                            try {
+                                            let receipt = await waitForReceipt(txHash);
+                                            if(receipt.status) {
+                                                resolve(receipt);
+                                            }
+                                        } catch(err) {
+                                            reject(err);
+                                        }
+                                        reject(new Error(`Execution failed <a target="_blank" href="${config.explorer}/tx/${txHash}">see Tx</a>`));
+                                    });
+                                } else {
+                                    resolve();
```

Comment:
Do we need the first already resolved: `Promise.resolve().then(() => { ...`
Why don't we just return (as line 1387) `return new Promise...`

Reply:
if i do that it does not work correctly, for some reason when the firs Promise resolves it triggers the second promise but then it goes straight for the third .then